File: src/main/java/code/vanilson/marketplace/model/Payment.java
package code.vanilson.marketplace.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Objects;

@Setter
@Getter
@Entity
@Table(name = "tb_payments")
@JsonPropertyOrder({"paymentId", "orderId", "customerId", "paymentMethod", "paymentStatus", "amount", "currency", "transactionId", "createdAt", "updatedAt", "version"})
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "payment_id", nullable = false)
    @JsonProperty("id")
    private Long paymentId;

    @Column(name = "order_id", nullable = false)
    private Long orderId;

    @Column(name = "customer_id", nullable = false)
    private Long customerId;

    @Column(name = "payment_method", nullable = false, length = 50)
    private String paymentMethod;

    @Column(name = "payment_status", nullable = false, length = 20)
    private String paymentStatus = "PENDING";

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal amount;

    @Column(length = 3)
    private String currency = "USD";

    @Column(name = "transaction_id", length = 100)
    private String transactionId;

    @Column(name = "payment_gateway_response", columnDefinition = "TEXT")
    private String paymentGatewayResponse;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @Version
    private Integer version;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public boolean isSuccessful() {
        return "COMPLETED".equals(paymentStatus);
    }

    public Payment() {
    }

    public Payment(Long orderId, Long customerId, String paymentMethod, BigDecimal amount) {
        this.orderId = orderId;
        this.customerId = customerId;
        this.paymentMethod = paymentMethod;
        this.amount = amount;
        this.paymentStatus = "PENDING";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Payment payment = (Payment) o;
        return Objects.equals(paymentId, payment.paymentId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(paymentId);
    }

    @Override
    public String toString() {
        return "Payment{" +
                "paymentId=" + paymentId +
                ", orderId=" + orderId +
                ", paymentMethod='" + paymentMethod + '\'' +
                ", paymentStatus='" + paymentStatus + '\'' +
                ", amount=" + amount +
                '}';
    }
}

File: frontend/src/api/types/payment.ts
export interface Payment {
  id: number;
  orderId: number;
  customerId: number;
  paymentMethod: string;
  paymentStatus: 'PENDING' | 'PROCESSING' | 'COMPLETED' | 'FAILED';
  amount: number;
  currency: string;
  transactionId: string;
  createdAt: string;
  updatedAt: string;
  version: number;
}

export interface PaymentMethod {
  id: number;
  customerId: number;
  methodType: string;
  provider?: string;
  isDefault: boolean;
  isActive: boolean;
  lastFourDigits: string;
  expiryMonth: string;
  expiryYear: string;
  createdAt?: string;
  updatedAt?: string;
  version?: number;
}

export interface ProcessPaymentRequest {
  orderId: number;
  customerId: number;
  paymentMethod: string;
  token?: string;
}

export interface CreatePaymentMethodRequest {
  customerId: number;
  methodType: string;
  provider?: string;
  lastFourDigits: string;
  expiryMonth: string;
  expiryYear: string;
  isDefault?: boolean;
}
